src/ripple/unl/tests/BasicNetwork.h

```diff
@@ -317,55 +396,274 @@ class BasicNetwork
             The clock is advanced by the
             specified duration.
 
-        @return `true` if any message was processed.
+        @return `true` if any messages remain.
     */
     template <class Period, class Rep>
     bool
-    run_for (std::chrono::duration<
+    step_for (std::chrono::duration<
         Period, Rep> const& amount);
+};
 
-private:
-    Derived&
-    derived()
+//------------------------------------------------------------------------------
+
+template <class Peer>
+BasicNetwork<Peer>::queue_type::queue_type(
+        qalloc const& alloc)
+    : alloc_ (alloc)
+{
+}
+
+template <class Peer>
+BasicNetwork<Peer>::queue_type::~queue_type()
+{
+    for(auto iter = by_when_.begin();
+        iter != by_when_.end();)
     {
-        return *static_cast<Derived*>(this);
+        auto m = &*iter;
+        ++iter;
```

Comment:
I don't understand why this loop is written this way. It looks like we could just do `for (auto& i : by_when_){ auto m = &i; ...}` What am I missing?